core/arch/riscv/plat-virt/conf.mk

```diff
@@ -30,6 +30,7 @@ $(call force,CFG_16550_UART,y)
 $(call force,CFG_RISCV_TIME_SOURCE_RDTIME,y)
 CFG_RISCV_MTIME_RATE ?= 10000000
 CFG_RISCV_SBI ?= y
+CFG_RISCV_WITH_M_MODE_SM ?= y
```

Comment:
In case of selecting inter-domains messaging-based solution, how we will select it ?
No plan yet to add our implementation, let's keep it for now.

Reply:
Maybe we can select it by device tree node in the future ?
Like linux optee driver does:
https://elixir.bootlin.com/linux/latest/source/drivers/tee/optee/smc_abi.c#L1427
```c
static optee_invoke_fn *get_invoke_func(struct device *dev)
{
	const char *method;

	pr_info("probing for conduit method.\n");

	if (device_property_read_string(dev, "method", &method)) {
		pr_warn("missing \"method\" property\n");
		return ERR_PTR(-ENXIO);
	}

	if (!strcmp("hvc", method))
		return optee_smccc_hvc;
	else if (!strcmp("smc", method))
		return optee_smccc_smc;

	pr_warn("invalid \"method\" property: %s\n", method);
	return ERR_PTR(-EINVAL);
}
```